### PR TITLE
[Polytope] Add basic plot recipe for polytopes

### PR DESCRIPTION
**Pull request by zerefwayne**

### CODE REVIEW

File: src/polytypes.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function default_theme(scene::SceneLike, ::Type{<: Plot(Polytope)})
```
Comment (juliohm) [inline src/polytypes.jl:1]: Extra space after `<:`

File: src/polytypes.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    Theme(
+        color = :red,
```
Comment (juliohm) [inline src/polytypes.jl:3]: Let's keep things simple, `color=:black` by default seems more neutral.

File: src/polytypes.jl [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    )
+end
+
+"""
+    Function to plot all types which are type Meshes.Polytype
+"""
+function AbstractPlotting.plot!(plot::Plot(Polytope))
+    pt = to_value(plot[1])
+    v = coordinates.(vertices(pt))
+    poly!(plot, v, color = :transparent, strokecolor = plot[:color], strokewidth = plot[:strokewidth])
```
Comment (juliohm) [inline src/polytypes.jl:14]: Awesome. As we discussed, it would be nice to replace `poly!` by some primitive recipe.

**Review by juliohm — CHANGES_REQUESTED** [review]
It is nice to see this work getting started. After you fix these issues, please consider adding a README with a simple example of usage.

**Comment by zerefwayne** [comment]
### 2D Polytope

image

### 3D Polytope

image

File: src/polytypes.jl [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    # push first element to create complete the polygon
+    # [(1, 0), (0, 0), (0, 1)] -> [(1, 0), (0, 0), (0, 1), (1, 0)]
+    v = Array.(v)
+    push!(v, v[1])
+    # extract x indices
+    xs = ((x) -> x[1]).(v)
+    # extract y indices
+    ys = ((x) -> x[2]).(v)
+    if embeddim(pt) == 2
+        lines!(plot, xs, ys)
```
Comment (juliohm) [inline src/polytypes.jl:25]: So is this the basic building block? isn't it `linesegments!`?

Also, I think we don't want to create segments like this. We want to take the edges of the polytope with `edges = facets(pt)` and then plot each segment:

```julia
for e in edges
  u, v = coordinates.(vertices(e))
  linesegments!(plot, u, v)
end
```
or something like that.

File: src/polytypes.jl [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    elseif embeddim(pt) == 3
+        # if third dimension exists, extract z indices
+        zs = ((x) -> x[3]).(v)
+        lines!(plot, xs, ys, zs)
+    end
```
Comment (juliohm) [inline src/polytypes.jl:30]: I think the code will be agnostic to the dimension if you use the linesegments approach we discussed some time ago. I don't understand why you used `lines!` here.

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/MeshesRecipes.jl [hunk 6]
```diff
@@ -1,5 +1,9 @@
 module MeshesRecipes
 
-# Write your package code here.
+using Meshes
+using AbstractPlotting
+import AbstractPlotting: Plot, default_theme, plot!, to_value
```
Comment (juliohm) [inline src/MeshesRecipes.jl:5]: Add a blank line before the import, it is nice to separate using from import statements

File: src/polytypes.jl [hunk 7]
```diff
@@ -0,0 +1,22 @@
+function default_theme(scene::SceneLike, ::Type{<:Plot(Polytope)})
+    Theme(
+        color = :black,
+        strokewidth = 2
+    )
+end
```
Comment (juliohm) [inline src/polytypes.jl:6]: Mind using 2 whitespaces for indentation to follow Meshes.jl conventions?

File: src/polytypes.jl [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+"""
+    Function to plot all types which are type Meshes.Polytope
+"""
```
Comment (juliohm) [inline src/polytypes.jl:10]: A simpler docstring would be:

>Function to plot general Polytope objects.

File: src/polytypes.jl [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+function AbstractPlotting.plot!(plot::Plot(Polytope))
+    pt = to_value(plot[1])
+    v = coordinates.(vertices(pt))
+    scatter!(plot, v, color = plot[:color])
+    edges = facets(pt)
+    for edge in edges
+        v = edge.vertices
+        vc = coordinates.(v)
+        lines!(plot, vc)
+    end
```
Comment (juliohm) [inline src/polytypes.jl:20]: We can generalize this logic to N-dimensional polytopes. You are typing "edge" and "vertices" etc, but this only applies to Polygon = Polytope{2}. In the case of a Polyhedra = Polygon{3} the facets are "areas" or triangles to be plotted.

Perhaps this function should be generalized with a recursion or you should break it into two methods one for 2D Polygon and another for 3D Polyhedra. Makes sense?

I like the code very much though, much cleaner than Plots.jl recipes in general.

**Review by juliohm — CHANGES_REQUESTED** [review]

File: src/polytypes.jl (hunk 9 above)
Comment (zerefwayne) [reply src/polytypes.jl:20]: For now, I tested polytopes in 2D & 3D spaces and the code functions well. I'll try to create a separate function for polyhedras in the next PR.

**Review by juliohm — APPROVED** [review]

**Pull request merged**